<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Alex Kuhn">

<title>MRcML Results</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="mrcml_data_analysis_files/libs/clipboard/clipboard.min.js"></script>
<script src="mrcml_data_analysis_files/libs/quarto-html/quarto.js"></script>
<script src="mrcml_data_analysis_files/libs/quarto-html/popper.min.js"></script>
<script src="mrcml_data_analysis_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="mrcml_data_analysis_files/libs/quarto-html/anchor.min.js"></script>
<link href="mrcml_data_analysis_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="mrcml_data_analysis_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="mrcml_data_analysis_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="mrcml_data_analysis_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="mrcml_data_analysis_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">MRcML Results</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Alex Kuhn </p>
          </div>
  </div>
    
  
    
  </div>
  

</header>

<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)  </span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(TwoSampleMR)             <span class="co"># Read in GWAS data; MR-IVW, MR-Egger,</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>                                 <span class="co"># MR-Weighted-Median, MR-Weighted-Mode, MR-RAPS</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(MRcML)                   <span class="co"># MRcML</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cause)                   <span class="co"># CAUSE</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(MendelianRandomization)  <span class="co"># MR-ContMix, MR-Lasso</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(MRMix)                   <span class="co"># MR-Mix</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(MRPRESSO)                <span class="co"># MR-PRESSO</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Load in own helper functions</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"mrcml_functions.R"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="plan" class="level3">
<h3 class="anchored" data-anchor-id="plan">Plan</h3>
<p>Here we will first reproduce the results from the data analysis in the MRcML paper in order to identify possible causal effects of 12 risk factors on three cardiometabolic diseases - coronary artery disease (CAD), stroke, and type 2 diabetes (T2D) - plus asthma largely as a negative control. These 12 risk factors (and their corresponding GWASs) are LDL cholesterol, HDL cholesterol, triglycerides (TG), drinks per week (alcohol), ever regular smoker (smoke), body fat percentage (BF), birth weight (BW), body mass index (BMI), height, fasting glucose (FG), systolic blood pressure (SBP), and diastolic blood pressure (DBP). See paper (pg. 9) for information on sample sizes.</p>
<p>For each risk factor/exposure-disease/outcome pair, we used the set of LD-independent SNPs as IVs as described in Morrison et al.&nbsp;using the groupings of (likely causal), correlated, unrelated, and non-causal.</p>
<p>I will attempt here to reproduce results to create Figure 8 and Figure 9 (without running CAUSAL for now) adding perhaps MR-Lasso and MR-PRESSO results.</p>
</section>
<section id="data-processing" class="level3">
<h3 class="anchored" data-anchor-id="data-processing">Data Processing</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Still a bit confused here... first task!</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="co"># ID of Fasting Proinsulin (Fasting glucose; FG?) as exposure, </span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="co"># ID of Age at Smoking as outcome </span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>exp_id <span class="ot">&lt;-</span> <span class="st">"ebi-a-GCST001212"</span> <span class="co"># FG</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>out_id <span class="ot">&lt;-</span> <span class="st">"ieu-a-964"</span>        <span class="co"># smoke</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract IVs for exposure</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>exposure_dat <span class="ot">&lt;-</span> <span class="fu">extract_instruments</span>(exp_id)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>API: public: http://gwas-api.mrcieu.ac.uk/</code></pre>
</div>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># LD-clumping of extracted IVs</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>exposure_dat <span class="ot">&lt;-</span> <span class="fu">clump_data</span>(exposure_dat)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Please look at vignettes for options on running this locally if you need to run many instances of this command.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Clumping ebi-a-GCST001212, 8 variants, using EUR population reference</code></pre>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract summary statistics of outcome, do not use proxy</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>outcome_dat <span class="ot">&lt;-</span> <span class="fu">extract_outcome_data</span>(exposure_dat<span class="sc">$</span>SNP, out_id, <span class="at">proxies =</span> <span class="dv">0</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Extracting data for 8 SNP(s) from 1 GWAS(s)</code></pre>
</div>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Harmonize data: correct strand for non-palindromic SNPs, drop all palindromic SNPs</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>dat <span class="ot">&lt;-</span> <span class="fu">harmonise_data</span>(exposure_dat, outcome_dat, <span class="at">action =</span> <span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Harmonising Proinsulin levels || id:ebi-a-GCST001212 (ebi-a-GCST001212) and Age of smoking initiation || id:ieu-a-964 (ieu-a-964)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Removing the following SNPs for being palindromic:
rs10501320, rs6235</code></pre>
</div>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>final_dat <span class="ot">&lt;-</span> dat[dat<span class="sc">$</span>mr_keep,]</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Get final estimates and standard errors for exposure and outcome</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>b_exp <span class="ot">&lt;-</span> final_dat<span class="sc">$</span>beta.exposure</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>b_out <span class="ot">&lt;-</span> final_dat<span class="sc">$</span>beta.outcome</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>se_exp <span class="ot">&lt;-</span> final_dat<span class="sc">$</span>se.exposure</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>se_out <span class="ot">&lt;-</span> final_dat<span class="sc">$</span>se.outcome</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Compare to built in function TwoSampleMR</span></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Looks like this does not do clumping, will proceed with own function</span></span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a><span class="co"># which I call: `exposure_outcome_processing`</span></span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a><span class="fu">make_dat</span>(<span class="st">"ebi-a-GCST001212"</span>, <span class="st">"ieu-a-964"</span>, <span class="at">proxies =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Extracting data for 8 SNP(s) from 1 GWAS(s)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Harmonising Proinsulin levels || id:ebi-a-GCST001212 (ebi-a-GCST001212) and Age of smoking initiation || id:ieu-a-964 (ieu-a-964)</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>         SNP effect_allele.exposure other_allele.exposure effect_allele.outcome
1 rs10501320                      C                     G                     C
2 rs11558471                      G                     A                     G
3 rs11603334                      A                     G                     A
4  rs4502156                      C                     T                     C
5  rs4790333                      T                     C                     T
6  rs6101962                      G                     A                     G
7     rs6235                      G                     C                     G
8  rs7903146                      T                     C                     T
  other_allele.outcome beta.exposure beta.outcome eaf.exposure eaf.outcome
1                    G        -0.110       0.0005        0.292      0.2835
2                    A        -0.057       0.0001        0.748      0.3015
3                    G         0.130      -0.0044        0.129      0.1505
4                    T        -0.048      -0.0008        0.580      0.4121
5                    C         0.040      -0.0029        0.398      0.4536
6                    A        -0.150       0.0009        0.893      0.1093
7                    C         0.058       0.0023        0.733      0.2827
8                    C         0.067      -0.0002        0.279      0.2998
  remove palindromic ambiguous id.outcome chr       pos se.outcome
1  FALSE        TRUE     FALSE  ieu-a-964  11  47293799     0.0026
2  FALSE       FALSE     FALSE  ieu-a-964   8 118185733     0.0025
3  FALSE       FALSE     FALSE  ieu-a-964  11  72432985     0.0032
4  FALSE       FALSE     FALSE  ieu-a-964  15  62383155     0.0023
5  FALSE       FALSE     FALSE  ieu-a-964  17   2262703     0.0023
6  FALSE       FALSE     FALSE  ieu-a-964  20  39021330     0.0084
7  FALSE        TRUE     FALSE  ieu-a-964   5  95728898     0.0026
8  FALSE       FALSE     FALSE  ieu-a-964  10 114758349     0.0025
  samplesize.outcome pval.outcome                                   outcome
1              47961       0.8315 Age of smoking initiation || id:ieu-a-964
2              47961       0.9816 Age of smoking initiation || id:ieu-a-964
3              47961       0.1736 Age of smoking initiation || id:ieu-a-964
4              47961       0.7198 Age of smoking initiation || id:ieu-a-964
5              47961       0.2020 Age of smoking initiation || id:ieu-a-964
6              47961       0.9178 Age of smoking initiation || id:ieu-a-964
7              47961       0.3777 Age of smoking initiation || id:ieu-a-964
8              47961       0.9478 Age of smoking initiation || id:ieu-a-964
       originalname.outcome                outcome.deprecated mr_keep.outcome
1 Age of smoking initiation Age of smoking initiation ||  ||             TRUE
2 Age of smoking initiation Age of smoking initiation ||  ||             TRUE
3 Age of smoking initiation Age of smoking initiation ||  ||             TRUE
4 Age of smoking initiation Age of smoking initiation ||  ||             TRUE
5 Age of smoking initiation Age of smoking initiation ||  ||             TRUE
6 Age of smoking initiation Age of smoking initiation ||  ||             TRUE
7 Age of smoking initiation Age of smoking initiation ||  ||             TRUE
8 Age of smoking initiation Age of smoking initiation ||  ||             TRUE
  data_source.outcome samplesize.exposure chr.exposure pos.exposure
1                 igd               10701           11     47293799
2                 igd               10701            8    118185733
3                 igd               10701           11     72432985
4                 igd               10701           15     62383155
5                 igd               10701           17      2262703
6                 igd               10701           20     39021330
7                 igd               10701            5     95728898
8                 igd               10701           10    114758349
  pval.exposure se.exposure      id.exposure
1   2.05400e-45      0.0080 ebi-a-GCST001212
2   4.21308e-13      0.0079 ebi-a-GCST001212
3   3.24863e-38      0.0100 ebi-a-GCST001212
4   3.80540e-11      0.0072 ebi-a-GCST001212
5   2.15601e-08      0.0071 ebi-a-GCST001212
6   9.89464e-12      0.0220 ebi-a-GCST001212
7   7.70726e-14      0.0078 ebi-a-GCST001212
8   3.48418e-18      0.0077 ebi-a-GCST001212
                                  exposure mr_keep.exposure
1 Proinsulin levels || id:ebi-a-GCST001212             TRUE
2 Proinsulin levels || id:ebi-a-GCST001212             TRUE
3 Proinsulin levels || id:ebi-a-GCST001212             TRUE
4 Proinsulin levels || id:ebi-a-GCST001212             TRUE
5 Proinsulin levels || id:ebi-a-GCST001212             TRUE
6 Proinsulin levels || id:ebi-a-GCST001212             TRUE
7 Proinsulin levels || id:ebi-a-GCST001212             TRUE
8 Proinsulin levels || id:ebi-a-GCST001212             TRUE
  pval_origin.exposure data_source.exposure action mr_keep
1             reported                  igd      1    TRUE
2             reported                  igd      1    TRUE
3             reported                  igd      1    TRUE
4             reported                  igd      1    TRUE
5             reported                  igd      1    TRUE
6             reported                  igd      1    TRUE
7             reported                  igd      1    TRUE
8             reported                  igd      1    TRUE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Own function </span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>FG_smoke <span class="ot">&lt;-</span> <span class="fu">exposure_outcome_processing</span>(<span class="at">exp_id =</span> <span class="st">"ebi-a-GCST001212"</span>, </span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">out_id =</span> <span class="st">"ieu-a-964"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Please look at vignettes for options on running this locally if you need to run many instances of this command.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Clumping ebi-a-GCST001212, 8 variants, using EUR population reference</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Extracting data for 8 SNP(s) from 1 GWAS(s)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Harmonising Proinsulin levels || id:ebi-a-GCST001212 (ebi-a-GCST001212) and Age of smoking initiation || id:ieu-a-964 (ieu-a-964)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Removing the following SNPs for being palindromic:
rs10501320, rs6235</code></pre>
</div>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>FG_smoke</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$b_exp
[1] -0.057  0.130 -0.048  0.040 -0.150  0.067

$b_out
[1]  0.0001 -0.0044 -0.0008 -0.0029  0.0009 -0.0002

$se_exp
[1] 0.0079 0.0100 0.0072 0.0071 0.0220 0.0077

$se_out
[1] 0.0025 0.0032 0.0023 0.0023 0.0084 0.0025</code></pre>
</div>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(FG_smoke)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       Length Class  Mode   
b_exp  6      -none- numeric
b_out  6      -none- numeric
se_exp 6      -none- numeric
se_out 6      -none- numeric</code></pre>
</div>
</div>
<p>Now that I know the code above works, I need the ID for each exposure and outcome of interest. How do I know which data set to choose? For example, at the host website: https://gwas.mrcieu.ac.uk/ it appears that there are four different data sets for CAD. There is:</p>
<ul>
<li>ebi-a-GCST005195</li>
<li>ebi-a-GCST005194</li>
<li>bbj-a-159</li>
<li>ebi-a-GCST003116</li>
</ul>
<p>Do I default to a certain year? The one with the largest sample size? Something else?</p>
<div class="cell">

</div>
</section>
<section id="possible-next-steps" class="level3">
<h3 class="anchored" data-anchor-id="possible-next-steps">Possible Next Steps</h3>
<p>There are a ton of results that I could reproduce here. Some that seem particularly helpful:</p>
<ul>
<li>It would be a good idea to reproduce the results in Table 1 and 2 in the supplementary materials (which has many more methods than above). After that, it would be a good idea to run the two GOF tests in some cases.</li>
<li>Worth time to reproduce simulation results? At least think about setup.</li>
<li>Some thoughts for <em>new</em> simulations/analyses if time permits: high dimensional setting (some results using profile likelihood), taking a data set with known population stratification issues and getting results, or simulating to see effect of post-selection inference carried out here in the GWAS analyses.</li>
</ul>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>